extensions/amp-twitter/0.1/amp-twitter.js

```diff
@@ -62,34 +93,67 @@ class AmpTwitter extends AMP.BaseElement {
     const iframe = getIframe(this.win, this.element, 'twitter', null,
         {allowFullscreen: true});
     this.applyFillContent(iframe);
-    listenFor(iframe, 'embed-size', data => {
-      // We only get the message if and when there is a tweet to display,
-      // so hide the placeholder
-      this.togglePlaceholder(false);
-      this./*OK*/changeHeight(data['height']);
+    iframe.classList.add('i-amp-html-twitter-iframe');
+
+    this.updateForLoadingState_();
+    listenFor(iframe, MessageType.EMBED_SIZE, data => {
+      this.updateForSuccessState_(data['height']);
     }, /* opt_is3P */true);
-    listenFor(iframe, 'no-content', () => {
-      const fallback = this.getFallback();
-      if (fallback) {
-        // If there is no content, but a fallback is provided.
-        this.togglePlaceholder(false);
-        this.toggleFallback(true);
-        this./*OK*/changeHeight(fallback./*OK*/offsetHeight);
-      } else {
-        // Else keep placeholder displayed since there's no fallback.
-        const placeholder = this.getPlaceholder();
-        if (placeholder) {
-          // Only happens if there is no content to render
-          // (e.g. tweet was deleted) and there is no fallback.
-          this./*OK*/changeHeight(placeholder./*OK*/offsetHeight);
-        }
-      }
+    listenFor(iframe, MessageType.NO_CONTENT, () => {
+      this.updateForFailureState_();
     }, /* opt_is3P */true);
     this.element.appendChild(iframe);
     this.iframe_ = iframe;
     return this.loadPromise(iframe);
   }
 
+  /**
+   * Updates when starting to load a tweet.
+   * @private
+   */
+  updateForLoadingState_() {
+    this.element.setAttribute('loading', 'start');
```

Comment:
this will escape the experiment flags and folks may start targeting it. maybe for now add a `i-amphtml-?` classname so they can't target in custom css yet.

Reply:
Looks like you can use i-amphtml for attributes as well (https://www.ampproject.org/docs/fundamentals/spec#class-and-tag-names), so I changed the attribute name.